### PR TITLE
test(Pagination): rewrite to testing library

### PR DESCRIPTION
**Pull request by mainframev**
<br/><br/><br/><url>LiveURL: https://orbit-test-pagination.surge.sh</url>

### CODE REVIEW

File: packages/orbit-components/src/Pagination/__tests__/index.test.js [hunk 1]
```diff
@@ -1,24 +1,41 @@
...
+
+    expect(screen.getByTestId(dataTest)).toBeInTheDocument();
+    userEvent.click(buttons.slice(-1)[0]);
+    expect(onPageChange).toHaveBeenCalled();
+  });
+
+  it("should have next button disabled", () => {
+    render(<Pagination dataTest="test" pageCount={9} selectedPage={9} onPageChange={() => {}} />);
+    const buttons = screen.getAllByRole("button");
+    expect(buttons.slice(-1)[0]).toHaveAttribute("disabled");
```
Comment (silvenon) [inline packages/orbit-components/src/Pagination/__tests__/index.test.js:34]:
```suggestion
    expect(buttons.slice(-1)[0]).toBeDisabled();
```

File: packages/orbit-components/src/Pagination/__tests__/index.test.js [hunk 2]
```diff
@@ -1,24 +1,41 @@
...
+  });
+
+  it("should have first button disabled", () => {
+    render(<Pagination dataTest="test" pageCount={6} selectedPage={1} onPageChange={() => {}} />);
+    expect(screen.getAllByRole("button")[0]).toHaveAttribute("disabled");
```
Comment (silvenon) [inline packages/orbit-components/src/Pagination/__tests__/index.test.js:39]:
```suggestion
    expect(screen.getAllByRole("button")[0]).toBeDisabled();
```

**Review by silvenon — CHANGES_REQUESTED** [review]

File: packages/orbit-components/src/Pagination/__tests__/index.test.js [hunk 3]
```diff
@@ -1,24 +1,41 @@
...
+        pageCount={pageCount}
+        selectedPage={selectedPage}
+        onPageChange={onPageChange}
+      />,
+    );
+
+    const buttons = screen.getAllByRole("button");
+
+    expect(screen.getByTestId(dataTest)).toBeInTheDocument();
+    userEvent.click(buttons.slice(-1)[0]);
```
[the remark this answers is not in the document]
Comment (mainframev) [reply packages/orbit-components/src/Pagination/__tests__/index.test.js:27]: ButtonLink supports `aria-label`, it has prop `title` which is passed to `ButtonPrimitive` and there is `aria-label={title}`

[the remark this answers is not in the document]
Comment (mainframev) [reply packages/orbit-components/src/Pagination/__tests__/index.test.js:27]: + i tried it with `{name: "Next"}` and it was not working, so i've added aria-labels via `title` prop, now test passes with `screen.getByLabelText`

[the remark this answers is not in the document]
Comment (silvenon) [reply packages/orbit-components/src/Pagination/__tests__/index.test.js:27]: This suggestion has grown outside the scope of this PR and has been moved to #2542.

**Comment by silvenon** [comment]
I assume you plan to rebase and update tests now that you can mock `useMediaQuery`?

**Comment by mainframev** [comment]
> I assume you plan to rebase and update tests now that you can mock `useMediaQuery`?

yes, sorry for not mentioning it

**Review by silvenon — APPROVED** [review]
Looks good!

In the next PR please also add tests for mobile, considering that you refactored the component with `useMediaQuery`. You can use Modal tests as a reference how to mock different breakpoints for different tests, just don't forget to restore at the end of each test so that the tests don't affect each other.

Also add a test to check whether passing `hideLabels` (short for `hideLabels={true}`) indeed hides the labels, and that `hideLabels={false}` shows them. You can check for that with `toHaveTextContent`.

**Pull request merged**